nemo/core/classes/common.py

```diff
@@ -45,12 +45,18 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
     def _validate_input_types(self, **kwargs):
         # TODO: Properly implement this
         if self.input_types is not None:
-            if len(kwargs) != len(self.input_types):
-                raise TypeError(
-                    "Number of input arguments provided ({}) is not as expected ({})".format(
-                        len(kwargs), len(self.input_types)
+            total_input_types = len(self.input_types)
+            mandatory_input_types = len(
+                [type_val for type_key, type_val in self.input_types.items() if not type_val.optional]
```

Comment:
How does one define that it is optional?

Reply:
NeuralType has the argument optional = False by default